actors/miner/src/lib.rs

```diff
@@ -4904,13 +4916,97 @@ fn confirm_sector_proofs_valid_internal(
 
         Ok((total_pledge, newly_vested))
     })?;
-
     // Request pledge update for activated sector.
     notify_pledge_changed(rt, &(total_pledge - newly_vested))?;
 
     Ok(())
 }
 
+pub struct SectorActivation {
+    sector_number: u64,
+    seal_proof: RegisteredSealProof,
+    expiration: ChainEpoch,
+    sealed_cid: Cid,
+    data_activation: DataActivation,
+    pre_commit_deposit: Option<TokenAmount>,
+}
+
+impl SectorActivation {
+    // XXX: if we stop putting deal ids on sector infos this
+    // function can go away before merging back to master
+    fn deal_ids(&self) -> Vec<DealID> {
+        match &self.data_activation {
+            BuiltinActivation(deal_ids) => deal_ids.clone(),
+            //_ => vec![],
+        }
+    }
+}
+
+impl From<SectorPreCommitOnChainInfo> for SectorActivation {
+    fn from(si: SectorPreCommitOnChainInfo) -> SectorActivation {
+        SectorActivation {
+            sector_number: si.info.sector_number,
+            seal_proof: si.info.seal_proof,
+            expiration: si.info.expiration,
+            sealed_cid: si.info.sealed_cid,
+            data_activation: BuiltinActivation(si.info.deal_ids), // XXX distinguish data activation type from existence of deal_id / commD when generalizing
+            pre_commit_deposit: Some(si.pre_commit_deposit),
+        }
+    }
+}
+
+/*
+    BuiltinMarket is today's call in CSPV f05 -> f06, return deal weights as today
+    BuiltinMarketWithCommD is what PRU should do to avoid double the market calls, return deal weights as today
+    SectorContentAdded is the new thing and can include piece manifests, independent calls f05 (or other) and f06, return deal weights per spec
```

Comment:
I'm not sure this should return deal weights (depending if you mean return from the f05, or from the inner activate_data method). The miner knows the size of the data items. The x10 verified weight multiplication is already done in the miner. The prior types need to return this from the market, but this one doesn't.

Reply:
While I'm moving away from a single function I want a general data activation interface that notifies sector consumers and returns deal weight info which is exactly what the miner actor needs for its separate sector info function.   It doesn't matter if the implementation of deal weight calculation is on another actor or inside the miner actor itself, Im asserting it belongs in the abstract data activation function.